lightning/src/ln/onion_utils.rs

```diff
@@ -446,10 +452,14 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(secp_ctx: &
 							});
 						}
 
+						if short_channel_id.is_none() {
```

Comment:
Oops, I don't think we want to set scid when `payment_failed` as its things like "unknown payment hash" which shouldn't factor into scoring (when they come from the last-hop). However, we'll need to exempt the 18|19 cases from this which imply someone on the route screwed with the payment, probably the second-to-last-hop.

Reply:
Ok, latest fixup should reflect this, IIUC. Think I'm getting a better feel for this code.